systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java

```diff
@@ -68,22 +70,32 @@ public LogCollector(String testSuiteName, String testCaseName, KubeClient kubeCl
 
         this.testSuite = new File(logSuiteDir);
 
-        // @BeforeAll in AbstractST always pass, and I ensure that we always deploy CO and after that some error can occur
-        this.clusterOperatorPod = kubeClient.getClient().pods().inAnyNamespace().list().getItems().stream()
-            .filter(pod -> pod.getMetadata().getName().contains(Constants.STRIMZI_DEPLOYMENT_NAME))
+        Optional<Pod> optionalCoPod = kubeClient.getClient().pods().inAnyNamespace().list().getItems().stream()
+            .filter(pod -> pod.getMetadata().getName().contains(ResourceManager.getCoDeploymentName()))
             // contract only one Cluster Operator deployment inside all namespaces
-            .findFirst()
-            .orElseThrow();
-        this.clusterOperatorNamespace = this.clusterOperatorPod.getMetadata().getNamespace();
+            .findFirst();
+
+        if (optionalCoPod.isPresent()) {
+            // @BeforeAll in AbstractST always pass, and I ensure that we always deploy CO and after that some error can occur
+            this.clusterOperatorPod = optionalCoPod.get();
+            this.clusterOperatorNamespace = this.clusterOperatorPod.getMetadata().getNamespace();
+        }
+
```

Comment:
I did quite a similar implementation here https://github.com/strimzi/strimzi-kafka-operator/pull/5137/files#diff-ec5e66aebd68442ff335612c7007ba4e3950625678a53d225b841179b5c1a83bL71-R80 to remove masking error when Co is not present